javagen/src/main/java/com/azure/autorest/util/MethodUtil.java

```diff
@@ -55,4 +70,155 @@ public static boolean isMethodIncludeRepeatableRequestHeaders(ProxyMethod proxyM
+        Set<String> mediaTypes = new HashSet<>();
+        for (Request request : requests) {
+            if (request.getProtocol().getHttp().getMediaTypes() != null
+                    && !request.getProtocol().getHttp().getMediaTypes().isEmpty()) {
+                mediaTypes.addAll(request.getProtocol().getHttp().getMediaTypes());
+            }
+        }
+        return mediaTypes.size() > 1;
+
+    }
+
+    /**
+     *
+     * @param request the request to put contentType on
+     * @param operation
+     * @return the created content type parameter
+     */
+    private static Parameter createContentTypeParameter(Request request, Operation operation) {
+        List<Request> requests = operation.getRequests();
+        Parameter contentType = new Parameter();
+        contentType.setOperation(operation);
+        contentType.setDescription("The content type");
+        for (int i = 0; i < request.getParameters().size(); ++i) {
+            if (request.getParameters().get(i).getSchema() instanceof BinarySchema) {
+                contentType.setRequired(request.getParameters().get(i).isRequired());
+                break;
+            }
+        }
+        contentType.setImplementation(Parameter.ImplementationLocation.METHOD);
+        SealedChoiceSchema sealedChoiceSchema = new SealedChoiceSchema();
+        StringSchema stringSchema = new StringSchema();
+        stringSchema.setType(Schema.AllSchemaTypes.STRING);
+        sealedChoiceSchema.setChoiceType(stringSchema);
+        sealedChoiceSchema.setChoices(getContentTypeChoiceValues(requests));
+        sealedChoiceSchema.setType(Schema.AllSchemaTypes.fromValue("sealed-choice"));
+        Language language = new Language();
+        language.setName("contentType");
+        language.setSerializedName("Content-Type");
+        language.setDescription("The content type");
+        sealedChoiceSchema.setLanguage(new Languages());
+        sealedChoiceSchema.getLanguage().setJava(language);
+        sealedChoiceSchema.getLanguage().setDefault(language);
+        sealedChoiceSchema.getLanguage().setJava(language);
+        sealedChoiceSchema.setProtocol(new Protocols());
+        contentType.setSchema(sealedChoiceSchema);
+        contentType.setImplementation(Parameter.ImplementationLocation.METHOD);
+        contentType.setProtocol(new Protocols());
+        contentType.getProtocol().setHttp(new Protocol());
+        contentType.getProtocol().getHttp().setIn(RequestParameterLocation.HEADER);
+        contentType.setLanguage(new Languages());
+        contentType.getLanguage().setDefault(language);
+        contentType.getLanguage().setJava(language);
+        return contentType;
+    }
+
+    /**
+     *
+     * @param parameters a list of parameters
+     * @return return the index of the BinarySchema parameter, if not found, return -1
+     */
+    private static int getBinarySchemaBodyOrContentLengthParameterIndex(List<Parameter> parameters) {
```

Comment:
This method name is revealing the implementation rather than the intent. Can we name this as `findIndexForContentTypeParam`?